packages/core/src/shared/telemetry/util.ts

```diff
@@ -203,3 +203,15 @@ export async function setupTelemetryId(extensionContext: vscode.ExtensionContext
         getLogger().error(`Erro while setting up telemetry id ${err}`)
     }
 }
+
+/**
+ * Potentially helpful values for the 'source' field in telemetry.
+ */
+export const ExtStartUpSources = {
+    firstStartUp: 'firstStartUp',
+    update: 'update',
+    reload: 'reload',
+    none: 'none',
+} as const
+
+export type ExtStartUpSource = (typeof ExtStartUpSources)[keyof typeof ExtStartUpSources]
```

Comment:
This pattern is useful if any code needs to *iterate* these values. Otherwise it's not necessary.